store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java

```diff
@@ -83,8 +105,26 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         }
 
         if (prefs.containsKey(Provisioning.A_zimbraPrefMailForwardingAddress)) {
-            if (!account.getBooleanAttr(Provisioning.A_zimbraFeatureMailForwardingEnabled, false))
+            if (!account.getBooleanAttr(Provisioning.A_zimbraFeatureMailForwardingEnabled, false)) {
                 throw ServiceException.PERM_DENIED("forwarding not enabled");
+            } else {
+                if (account.getBooleanAttr(
+                    Provisioning.A_zimbraFeatureAddressVerificationEnabled, false)) {
+                    /*
+                     * forwarding address verification enabled, store the email
+                     * ID in 'zimbraFeatureMailForwardingVerificationAddress'
+                     * till the time it's verified
+                     */
+                    String emailIdToVerify = (String) prefs
+                        .get(Provisioning.A_zimbraPrefMailForwardingAddress);
+                    prefs.remove(Provisioning.A_zimbraPrefMailForwardingAddress);
+                    StringUtil.addToMultiMap(prefs,
+                        Provisioning.A_zimbraFeatureAddressUnderVerification,
+                        emailIdToVerify);
+                    Account authAccount = getAuthenticatedAccount(zsc);
```

Comment:
once the verification mail is sent, status should be updated to "Pending"

Reply:
done